In [2]:
#libraries
import requests
import pandas as pd

In [3]:
#
soil_data=pd.read_csv('soil_data.csv')

In [4]:
#soil_data

In [5]:
def get_meteo_csv(lat,lon,year,id,key):
    #extract meteo information for that year from 1 Marcg to 30 September in csv format
    url="https://weather.visualcrossing.com/VisualCrossingWebServices/rest/services/timeline/"+str(lat)+"%2C%20"+str(lon)+"/"+str(year)+"-03-15/"+str(year)+"-09-30?unitGroup=metric&include=days&key="+key+"&contentType=csv"
    data=requests.get(url)
    
    name="meteo/"+str(id)+".csv"
    csv_file = open(name, 'wb')

    csv_file.write(data.content)
    csv_file.close()

In [12]:
row=18

key="W7NLYHBGY68VPU9JCKA8HVEHS"

lat=soil_data.iloc[row]['latitudine']
lon=soil_data.iloc[row]['longitudine']
year=soil_data.iloc[row]['year']
id=soil_data.iloc[row]['id']
id

141393

In [24]:
#soil_data.iloc[18]['year']

0


## RIVEDI LA RIGA 18 con id=141393 --> certe osservazioni non avevano l'anno e l'ho messo a mano

## REDUCE METEO INFORMATION

In [11]:

#id=soil_data.iloc[row]['id']
name="meteo/"+str(id)+".csv"
name

'meteo/137591.csv'

In [12]:
current_data=pd.read_csv(name)

In [13]:
current_data.columns

Index(['name', 'datetime', 'tempmax', 'tempmin', 'temp', 'feelslikemax',
       'feelslikemin', 'feelslike', 'dew', 'humidity', 'precip', 'precipprob',
       'precipcover', 'preciptype', 'snow', 'snowdepth', 'windgust',
       'windspeed', 'winddir', 'sealevelpressure', 'cloudcover', 'visibility',
       'solarradiation', 'solarenergy', 'uvindex', 'severerisk', 'sunrise',
       'sunset', 'moonphase', 'conditions', 'description', 'icon', 'stations'],
      dtype='object')

In [14]:
to_eliminate=['name','datetime','feelslikemax','feelslikemin', 'feelslike','precipprob',
       'precipcover', 'preciptype','snow','snowdepth', 'windgust','winddir', 'sealevelpressure', 'cloudcover', 'visibility',
       'severerisk', 'sunrise',
       'sunset', 'moonphase', 'conditions', 'description', 'icon', 'stations']

In [15]:
current_data.drop(to_eliminate, inplace=True, axis=1)

In [19]:
current_data


tempmax  tempmin  temp   dew  humidity  precip  windspeed  \
0       15.9      3.3   9.0   2.4      65.2     0.0       16.5   
1       16.9      3.3  10.0   2.4      61.2     0.0       15.5   
2       21.7      2.4  12.3   2.4      56.1     0.0       16.3   
3       25.9     11.2  17.5   3.2      43.9     0.0       28.7   
4       18.0      8.2  14.1   8.2      69.2     0.0       16.1   
..       ...      ...   ...   ...       ...     ...        ...   
195     21.1     13.0  16.4  13.5      83.8     0.0       16.6   
196     23.0     12.0  17.6  13.6      79.0     0.0       16.9   
197     23.0     10.5  17.4  12.4      74.0     0.0       24.3   
198     22.0     10.2  16.6  11.5      73.4     0.1       16.6   
199     21.0     11.7  16.4  10.6      70.5     0.0       24.1   

     solarradiation  solarenergy  uvindex  
0             210.1         18.1        7  
1             230.3         20.1        8  
2             235.3         20.3        8  
3             229.2         19.9        7  
4             224.6         19.5        7  
..              ...          ...      ...  
195           155.8         13.4        6  
196           200.9         17.3        7  
197           205.2         17.9        7  
198           212.1         18.1        7  
199           217.2         18.4        7  

[200 rows x 10 columns]

In [26]:
tempmax=current_data['tempmax'].mean()
tempmin=current_data['tempmin'].mean()
temp=current_data['temp'].mean()
dew=current_data['dew'].mean()
humidity=current_data['humidity'].mean()
precip=current_data['precip'].sum()
windspeed=current_data['windspeed'].mean()
solarradiation=current_data['solarradiation'].mean()
solarenergy=current_data['solarenergy'].mean()
uvindex=current_data['uvindex'].mean()

In [27]:
def reduce_csv(row):
   #give back the reduced information for a meteo csv file
   #in order:tempmax,tempmin,temp,dew,humidity,precip,windspeed,solarradiation,solarenergy,uvindex
   id=soil_data.iloc[row]['id']
   name="meteo/"+str(id)+".csv"    
   current_data=pd.read_csv(name)

   #columns to eliminate
   to_eliminate=['name','datetime','feelslikemax','feelslikemin', 'feelslike','precipprob',
      'precipcover', 'preciptype','snow','snowdepth', 'windgust','winddir', 'sealevelpressure', 
      'cloudcover', 'visibility','severerisk', 'sunrise','sunset', 'moonphase', 'conditions', 
      'description', 'icon', 'stations']

   current_data.drop(to_eliminate, inplace=True, axis=1)

   tempmax=current_data['tempmax'].mean()
   tempmin=current_data['tempmin'].mean()
   temp=current_data['temp'].mean()
   dew=current_data['dew'].mean()
   humidity=current_data['humidity'].mean()
   precip=current_data['precip'].sum()
   windspeed=current_data['windspeed'].mean()
   solarradiation=current_data['solarradiation'].mean()
   solarenergy=current_data['solarenergy'].mean()
   uvindex=current_data['uvindex'].mean()

   return([tempmax,tempmin,temp,dew,humidity,precip,windspeed,solarradiation,solarenergy,uvindex])

In [28]:
row=0
tempmax,tempmin,temp,dew,humidity,precip,windspeed,solarradiation,solarenergy,uvindex=reduce_csv(row)

#### Attach this data to soil_data.csv 